src/main/react/src/views/Profile.js

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect } from "react";
+import {
+  CCard,
+  CCardBody,
+  CRow,
+  CCol,
+  CImg,
+  CCardHeader,
+  CNav,
+  CNavItem,
+  CNavLink,
+  CButton,
+  CBadge,
+} from "@coreui/react";
+import Select from "react-select";
+import Feed from "./Feed";
+import requestRoute, { getTags } from "../requests";
+import app from "firebase/app";
+import "firebase/auth";
+
+const Profile = () => {
+  const [signedIn, setSignedIn] = useState(false);
+  const [profilePic, setProfilePic] = useState(null);
+
+  const [feedType, setFeedType] = useState("saved");
+
+  const [allTags, setAllTags] = useState({});
+  const [selectedTagId, setSelectedTagId] = useState(null);
+  const [followedTagIds, setFollowedTagIds] = useState([]);
+
+  const followTag = async (user, tagId) => {
+    if (!followedTagIds.includes(tagId)) {
+      setFollowedTagIds(followedTagIds.concat(tagId));
+      let idToken = await user.getIdToken();
+      fetch(requestRoute + "api/user?type=FOLLOW_TAG&tagID=" + tagId + "&token=" + idToken, { method: "PUT" });
+    }
+  };
+
+  const unfollowTag = async (user, tagId) => {
+    if (followedTagIds.includes(tagId)) {
+      setFollowedTagIds(followedTagIds.filter(id => id !== tagId));
+      let idToken = await user.getIdToken();
+      fetch(requestRoute + "api/user?type=UNFOLLOW_TAG&tagID=" + tagId + "&token=" + idToken, { method: "PUT" });
+    }
+  };
+
+  const getFollowedTags = async user => {
+    let idToken = await user.getIdToken();
+    let tagsJson = await fetch(requestRoute + "api/user?type=FOLLOW_TAG&token=" + idToken);
+    let tagsData = await tagsJson.json();
+    return tagsData;
+  };
+
+  let listener = app.auth().onAuthStateChanged(user => {
+    if (user) {
+      listener(); // unsubscribe from this listener
+      setSignedIn(true);
+      setProfilePic(user.providerData[0].photoURL);
+    }
+  });
+
+  useEffect(() => {
+    getTags().then(setAllTags);
+    let listener2 = app.auth().onAuthStateChanged(async user => getFollowedTags(user).then(setFollowedTagIds));
+    return () => {
+      listener2();
+      setFollowedTagIds([]);
+      setAllTags({});
+    };
+  }, []);
+
+  return (
```

Comment:
Would be nice to have a place holder message when you don't have any saved/created recipes.

Reply:
Yup, done.